# Contexto

- Dados de vendas de lojas ficticias;
- Arquivo __Vendas.xlsx__ : dados de vendas diárias de janeiro até o mês de novembre do 2019;
- Arquivo __Vendas - Dez.xlsx__ : dados de vendas do mês de dezembro 2019;
- Arquivo __Gerentes.xlsx__ : dados dos gerentes das lojas.

## Tratamento dos dados (até novembro)

In [1]:
# importa as dependencias
import pandas as pd

In [2]:
# carrega os dados
vendas = pd.read_excel('Vendas.xlsx')
vendas.head()

Código Venda       Data             ID Loja         Produto  Quantidade  \
0             1 2019-01-01  Iguatemi Esplanada  Sapato Estampa           1   
1             1 2019-01-01  Iguatemi Esplanada        Camiseta           2   
2             1 2019-01-01  Iguatemi Esplanada   Sapato Xadrez           1   
3             2 2019-01-02      Norte Shopping         Relógio           3   
4             2 2019-01-02      Norte Shopping    Chinelo Liso           1   

   Valor Unitário  Valor Final  
0             358          358  
1             180          360  
2             368          368  
3             200          600  
4              71           71

In [3]:
# verifica a consistência dos dados
vendas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93910 entries, 0 to 93909
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Código Venda    93910 non-null  int64         
 1   Data            93910 non-null  datetime64[ns]
 2   ID Loja         93910 non-null  object        
 3   Produto         93910 non-null  object        
 4   Quantidade      93910 non-null  int64         
 5   Valor Unitário  93910 non-null  int64         
 6   Valor Final     93910 non-null  int64         
dtypes: datetime64[ns](1), int64(4), object(2)
memory usage: 5.0+ MB


In [4]:
# otimiza uso da memoria
vendas['ID Loja'] = vendas['ID Loja'].str.strip().astype('category')
vendas['Produto'] = vendas['Produto'].str.strip().astype('category')
vendas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93910 entries, 0 to 93909
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Código Venda    93910 non-null  int64         
 1   Data            93910 non-null  datetime64[ns]
 2   ID Loja         93910 non-null  category      
 3   Produto         93910 non-null  category      
 4   Quantidade      93910 non-null  int64         
 5   Valor Unitário  93910 non-null  int64         
 6   Valor Final     93910 non-null  int64         
dtypes: category(2), datetime64[ns](1), int64(4)
memory usage: 3.8 MB


## Análise de dados

### P1 - Analise as colunas Quantidade, Valor unitário e Valor Final

In [5]:
# visualiza estatistica das colunas
vendas[['Quantidade', 'Valor Unitário', 'Valor Final']].describe()

Quantidade  Valor Unitário   Valor Final
count  93910.000000    93910.000000  93910.000000
mean       2.166553      191.725886    414.862656
std        1.258732      145.215519    434.846228
min        1.000000       30.000000     30.000000
25%        1.000000      100.000000    156.000000
50%        2.000000      155.000000    274.000000
75%        3.000000      248.000000    524.000000
max        5.000000      750.000000   3750.000000

- Em média as pessoas compram 2 produtos por transação;
- Em média o valor dos produtos é de 192 reais;
- Em média o faturamento em cada transação é de 415 reais;
- Vende-se máximo por transação 5 unidades de um produto;
- O faturamento máximo de uma transação é de 3750 reais;
- O produto mais barato custa 30 reais.

### P2 - Qual o produto mais e menos vendido?

In [6]:
vendas[['Produto', 'Quantidade']].groupby(by='Produto').sum().sort_values(by='Quantidade', ascending=False)

Quantidade
Produto                        
Bermuda Linho              1893
Mochila Liso               1868
Short                      1844
Casaco Liso                1840
Cueca                      1837
...                         ...
Camisa Gola V Linho        1597
Meia Xadrez                1584
Gorro Estampa              1574
Sunga Estampa              1537
Short Listrado             1529

[120 rows x 1 columns]

- O produto mais vendido é Bermuda Linho;
- O produto menos vendido é Short Listrado

### P3 - Qual o produto da loja "Bourbon Shopping SP" que possui maior valor unitário?

In [7]:
# filtra por loja
lojaBouShoSP = vendas[vendas['ID Loja'] == 'Bourbon Shopping SP']
# lista os produtos desde o maior até o menor valor unitário
lojaBouShoSP[['Produto', 'Valor Unitário']].groupby(by='Produto').max().sort_values('Valor Unitário', ascending=False)

Valor Unitário
Produto                       
Terno Linho                750
Terno Liso                 720
Terno Listrado             709
Terno Xadrez               709
Terno Estampa              706
...                        ...
Cueca                       50
Meia Xadrez                 41
Meia Liso                   38
Meia Listrado               37
Meia                        30

[120 rows x 1 columns]

- O produto "Terno Linho" possui o maior valor unitário igual a 750 reais.

### P4 - Quanto custa o décimo produto mais caro na loja "Bourbon Shopping SP"?

In [8]:
# filtra por loja
lojaBouShoSP = vendas[vendas['ID Loja'] == 'Bourbon Shopping SP']
# lista os produtos desde o maior até o menor valor unitário
lojaBouShoSP[['Produto', 'Valor Unitário']].groupby(by='Produto').max().sort_values('Valor Unitário', ascending=False).iloc[9]

Valor Unitário    363
Name: Sapato Listrado, dtype: int64

- O custo do décimo produto mais caro na loja "Bourbon Shopping SP" é de 363 reais.

### P5 - Quanto custa o  produto "Chinelo Liso" na loja "Norte Shopping"?

In [9]:
# filtra por loja
lojaChLi = vendas[vendas['ID Loja'] == "Norte Shopping"]
# agrupa por produto
lojaChLi[['Produto', 'Valor Unitário']].groupby(by='Produto').max().loc['Chinelo Liso']

Valor Unitário    71
Name: Chinelo Liso, dtype: int64

- O produto "Chinelo Liso" custa 71 reais.

### P6 - Quais foram as cinco primeiras transações na loja "Rio Mar Shopping Fortaleza"?	

In [10]:
vendas[vendas['ID Loja'] == 'Rio Mar Shopping Fortaleza'].sort_values('Data', ascending=True).reset_index(drop=True).loc[:4]

Código Venda       Data                     ID Loja           Produto  \
0             3 2019-01-02  Rio Mar Shopping Fortaleza       Cinto Linho   
1           181 2019-01-02  Rio Mar Shopping Fortaleza       Cueca Linho   
2           179 2019-01-02  Rio Mar Shopping Fortaleza  Mochila Listrado   
3           150 2019-01-02  Rio Mar Shopping Fortaleza   Chinelo Estampa   
4           149 2019-01-02  Rio Mar Shopping Fortaleza     Camisa Gola V   

   Quantidade  Valor Unitário  Valor Final  
0           1             248          248  
1           2              93          186  
2           3             275          825  
3           5              74          370  
4           3             100          300

### P7 - Quais foram as cinco últimas transações na loja "Norte Shopping"?	

In [11]:
vendas.loc[vendas['ID Loja'] == 'Norte Shopping'].reset_index(drop=True).loc[(3924-5):]

Código Venda       Data         ID Loja         Produto  Quantidade  \
3919         64888 2019-11-30  Norte Shopping     Polo Xadrez           1   
3920         64914 2019-11-30  Norte Shopping      Calça Liso           1   
3921         64914 2019-11-30  Norte Shopping  Casaco Estampa           2   
3922         64914 2019-11-30  Norte Shopping           Gorro           2   
3923         64980 2019-11-30  Norte Shopping   Bermuda Linho           1   

      Valor Unitário  Valor Final  
3919             142          142  
3920             190          190  
3921             256          512  
3922              80          160  
3923             195          195

### P8 - Na loja "Norte Shopping", quais foram os 5 produtos mais vendidos por transação?	

In [12]:
vendas.loc[vendas['ID Loja'] == 'Norte Shopping', ['ID Loja', 'Produto', 'Quantidade']].sort_values(by='Quantidade', ascending=False).reset_index(drop=True).loc[0:4, 'Produto']

0    Pulseira Estampa
1               Sunga
2         Sapato Liso
3           Polo Liso
4        Bermuda Liso
Name: Produto, dtype: category
Categories (120, object): ['Bermuda', 'Bermuda Estampa', 'Bermuda Linho', 'Bermuda Liso', ..., 'Tênis Linho', 'Tênis Liso', 'Tênis Listrado', 'Tênis Xadrez']

### P9 - Adicione a coluna "Comissão" que é calculada como o 5% do "Valor Final"

In [13]:
vendas['Comissao'] = vendas['Valor Final']*.05
vendas

Código Venda       Data              ID Loja            Produto  \
0                 1 2019-01-01   Iguatemi Esplanada     Sapato Estampa   
1                 1 2019-01-01   Iguatemi Esplanada           Camiseta   
2                 1 2019-01-01   Iguatemi Esplanada      Sapato Xadrez   
3                 2 2019-01-02       Norte Shopping            Relógio   
4                 2 2019-01-02       Norte Shopping       Chinelo Liso   
...             ...        ...                  ...                ...   
93905         65012 2019-11-30  Shopping Vila Velha     Mochila Xadrez   
93906         65013 2019-11-30    Ribeirão Shopping  Pulseira Listrado   
93907         65013 2019-11-30    Ribeirão Shopping     Cueca Listrado   
93908         65014 2019-11-30     Shopping Morumbi     Pulseira Linho   
93909         65014 2019-11-30     Shopping Morumbi      Casaco Xadrez   

       Quantidade  Valor Unitário  Valor Final  Comissao  
0               1             358          358     17.90  
1               2             180          360     18.00  
2               1             368          368     18.40  
3               3             200          600     30.00  
4               1              71           71      3.55  
...           ...             ...          ...       ...  
93905           2             283          566     28.30  
93906           2              79          158      7.90  
93907           3              67          201     10.05  
93908           5             114          570     28.50  
93909           4             259         1036     51.80  

[93910 rows x 8 columns]

## Tratamento dos dados (dezembro)

In [14]:
# carrega os dados
vendas_dez = pd.read_excel('Vendas - Dez.xlsx')
vendas_dez

Código Venda       Data                     ID Loja           Produto  \
0            65014 2019-12-01            Shopping Morumbi    Sunga Listrado   
1            65014 2019-12-01            Shopping Morumbi   Casaco Listrado   
2            65016 2019-12-01           Iguatemi Campinas   Sapato Listrado   
3            65016 2019-12-01           Iguatemi Campinas            Casaco   
4            65017 2019-12-01          Shopping SP Market        Gorro Liso   
...            ...        ...                         ...               ...   
7084         69996 2019-12-26  Center Shopping Uberlândia    Short Listrado   
7085         69996 2019-12-26  Center Shopping Uberlândia           Mochila   
7086         69996 2019-12-26  Center Shopping Uberlândia  Pulseira Estampa   
7087         69997 2019-12-26           Ribeirão Shopping   Camisa Listrado   
7088         69997 2019-12-26           Ribeirão Shopping       Short Linho   

      Quantidade  Valor Unitário  Valor Final  
0              5             114          570  
1              1             269          269  
2              2             363          726  
3              1             250          250  
4              3              92          276  
...          ...             ...          ...  
7084           2             102          204  
7085           4             270         1080  
7086           1              87           87  
7087           1             108          108  
7088           2             133          266  

[7089 rows x 7 columns]

In [15]:
# verifica a consistências dos dados
vendas_dez.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7089 entries, 0 to 7088
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Código Venda    7089 non-null   int64         
 1   Data            7089 non-null   datetime64[ns]
 2   ID Loja         7089 non-null   object        
 3   Produto         7089 non-null   object        
 4   Quantidade      7089 non-null   int64         
 5   Valor Unitário  7089 non-null   int64         
 6   Valor Final     7089 non-null   int64         
dtypes: datetime64[ns](1), int64(4), object(2)
memory usage: 387.8+ KB


In [16]:
# otimiza os dados
vendas_dez['ID Loja'] = vendas_dez['ID Loja'].astype('category')
vendas_dez['Produto'] = vendas_dez['Produto'].astype('category')
vendas_dez.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7089 entries, 0 to 7088
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Código Venda    7089 non-null   int64         
 1   Data            7089 non-null   datetime64[ns]
 2   ID Loja         7089 non-null   category      
 3   Produto         7089 non-null   category      
 4   Quantidade      7089 non-null   int64         
 5   Valor Unitário  7089 non-null   int64         
 6   Valor Final     7089 non-null   int64         
dtypes: category(2), datetime64[ns](1), int64(4)
memory usage: 296.6 KB


### P10 - Junte os dados do mês de dezembro

In [17]:
vendasTot = pd.concat([vendas, vendas_dez], ignore_index=True)
vendasTot

Código Venda       Data                     ID Loja           Produto  \
0                  1 2019-01-01          Iguatemi Esplanada    Sapato Estampa   
1                  1 2019-01-01          Iguatemi Esplanada          Camiseta   
2                  1 2019-01-01          Iguatemi Esplanada     Sapato Xadrez   
3                  2 2019-01-02              Norte Shopping           Relógio   
4                  2 2019-01-02              Norte Shopping      Chinelo Liso   
...              ...        ...                         ...               ...   
100994         69996 2019-12-26  Center Shopping Uberlândia    Short Listrado   
100995         69996 2019-12-26  Center Shopping Uberlândia           Mochila   
100996         69996 2019-12-26  Center Shopping Uberlândia  Pulseira Estampa   
100997         69997 2019-12-26           Ribeirão Shopping   Camisa Listrado   
100998         69997 2019-12-26           Ribeirão Shopping       Short Linho   

        Quantidade  Valor Unitário  Valor Final  Comissao  
0                1             358          358     17.90  
1                2             180          360     18.00  
2                1             368          368     18.40  
3                3             200          600     30.00  
4                1              71           71      3.55  
...            ...             ...          ...       ...  
100994           2             102          204       NaN  
100995           4             270         1080       NaN  
100996           1              87           87       NaN  
100997           1             108          108       NaN  
100998           2             133          266       NaN  

[100999 rows x 8 columns]

### P11 - Exclua a coluna "Comissao"

In [18]:
# exclui coluna
vendasTot.drop('Comissao', axis=1)

Código Venda       Data                     ID Loja           Produto  \
0                  1 2019-01-01          Iguatemi Esplanada    Sapato Estampa   
1                  1 2019-01-01          Iguatemi Esplanada          Camiseta   
2                  1 2019-01-01          Iguatemi Esplanada     Sapato Xadrez   
3                  2 2019-01-02              Norte Shopping           Relógio   
4                  2 2019-01-02              Norte Shopping      Chinelo Liso   
...              ...        ...                         ...               ...   
100994         69996 2019-12-26  Center Shopping Uberlândia    Short Listrado   
100995         69996 2019-12-26  Center Shopping Uberlândia           Mochila   
100996         69996 2019-12-26  Center Shopping Uberlândia  Pulseira Estampa   
100997         69997 2019-12-26           Ribeirão Shopping   Camisa Listrado   
100998         69997 2019-12-26           Ribeirão Shopping       Short Linho   

        Quantidade  Valor Unitário  Valor Final  
0                1             358          358  
1                2             180          360  
2                1             368          368  
3                3             200          600  
4                1              71           71  
...            ...             ...          ...  
100994           2             102          204  
100995           4             270         1080  
100996           1              87           87  
100997           1             108          108  
100998           2             133          266  

[100999 rows x 7 columns]

### P11 - Exclua a primiera linha

In [19]:
vendasTot.drop(0, axis=0)

Código Venda       Data                     ID Loja           Produto  \
1                  1 2019-01-01          Iguatemi Esplanada          Camiseta   
2                  1 2019-01-01          Iguatemi Esplanada     Sapato Xadrez   
3                  2 2019-01-02              Norte Shopping           Relógio   
4                  2 2019-01-02              Norte Shopping      Chinelo Liso   
5                  3 2019-01-02  Rio Mar Shopping Fortaleza       Cinto Linho   
...              ...        ...                         ...               ...   
100994         69996 2019-12-26  Center Shopping Uberlândia    Short Listrado   
100995         69996 2019-12-26  Center Shopping Uberlândia           Mochila   
100996         69996 2019-12-26  Center Shopping Uberlândia  Pulseira Estampa   
100997         69997 2019-12-26           Ribeirão Shopping   Camisa Listrado   
100998         69997 2019-12-26           Ribeirão Shopping       Short Linho   

        Quantidade  Valor Unitário  Valor Final  Comissao  
1                2             180          360     18.00  
2                1             368          368     18.40  
3                3             200          600     30.00  
4                1              71           71      3.55  
5                1             248          248     12.40  
...            ...             ...          ...       ...  
100994           2             102          204       NaN  
100995           4             270         1080       NaN  
100996           1              87           87       NaN  
100997           1             108          108       NaN  
100998           2             133          266       NaN  

[100998 rows x 8 columns]

### P12 - Exclua as 10 primeiras linhas

In [20]:
vendasTot.drop([r for r in range(10)], axis=0)

Código Venda       Data                           ID Loja  \
10                 8 2019-01-02                 Iguatemi Campinas   
11                 9 2019-01-02  Shopping Center Leste Aricanduva   
12                 9 2019-01-02  Shopping Center Leste Aricanduva   
13                10 2019-01-02         Passei das Águas Shopping   
14                10 2019-01-02         Passei das Águas Shopping   
...              ...        ...                               ...   
100994         69996 2019-12-26        Center Shopping Uberlândia   
100995         69996 2019-12-26        Center Shopping Uberlândia   
100996         69996 2019-12-26        Center Shopping Uberlândia   
100997         69997 2019-12-26                 Ribeirão Shopping   
100998         69997 2019-12-26                 Ribeirão Shopping   

                 Produto  Quantidade  Valor Unitário  Valor Final  Comissao  
10         Terno Estampa           5             706         3530    176.50  
11           Tênis Linho           1             294          294     14.70  
12         Casaco Xadrez           2             259          518     25.90  
13       Camiseta Xadrez           2             200          400     20.00  
14            Cueca Liso           1              69           69      3.45  
...                  ...         ...             ...          ...       ...  
100994    Short Listrado           2             102          204       NaN  
100995           Mochila           4             270         1080       NaN  
100996  Pulseira Estampa           1              87           87       NaN  
100997   Camisa Listrado           1             108          108       NaN  
100998       Short Linho           2             133          266       NaN  

[100989 rows x 8 columns]

### P12 - Exclua as 3 primeiras colunas

In [29]:
vendasTot.drop(vendas.columns[:3], axis=1)

Produto  Quantidade  Valor Unitário  Valor Final  Comissao
0         Sapato Estampa           1             358          358     17.90
1               Camiseta           2             180          360     18.00
2          Sapato Xadrez           1             368          368     18.40
3                Relógio           3             200          600     30.00
4           Chinelo Liso           1              71           71      3.55
...                  ...         ...             ...          ...       ...
100994    Short Listrado           2             102          204       NaN
100995           Mochila           4             270         1080       NaN
100996  Pulseira Estampa           1              87           87       NaN
100997   Camisa Listrado           1             108          108       NaN
100998       Short Linho           2             133          266       NaN

[100999 rows x 5 columns]

### P13 - Adicione uma nova coluna vazia

In [30]:
vendasTot.loc[:, 'Vazia'] = None
vendasTot.head(3)

Código Venda       Data             ID Loja         Produto  Quantidade  \
0             1 2019-01-01  Iguatemi Esplanada  Sapato Estampa           1   
1             1 2019-01-01  Iguatemi Esplanada        Camiseta           2   
2             1 2019-01-01  Iguatemi Esplanada   Sapato Xadrez           1   

   Valor Unitário  Valor Final  Comissao Vazia  
0             358          358      17.9  None  
1             180          360      18.0  None  
2             368          368      18.4  None

### P14 - Deleta a coluna vazia adicionada anteriormente

In [32]:
vendasTot.dropna(how='all', axis=1, inplace=True)
vendasTot.head(3)

Código Venda       Data             ID Loja         Produto  Quantidade  \
0             1 2019-01-01  Iguatemi Esplanada  Sapato Estampa           1   
1             1 2019-01-01  Iguatemi Esplanada        Camiseta           2   
2             1 2019-01-01  Iguatemi Esplanada   Sapato Xadrez           1   

   Valor Unitário  Valor Final  Comissao  
0             358          358      17.9  
1             180          360      18.0  
2             368          368      18.4

### P15 - Deleta linhas completamente vazias

In [35]:
vendasTot.dropna(how='all', axis=0, inplace=True)
vendasTot

Código Venda       Data                     ID Loja           Produto  \
0                  1 2019-01-01          Iguatemi Esplanada    Sapato Estampa   
1                  1 2019-01-01          Iguatemi Esplanada          Camiseta   
2                  1 2019-01-01          Iguatemi Esplanada     Sapato Xadrez   
3                  2 2019-01-02              Norte Shopping           Relógio   
4                  2 2019-01-02              Norte Shopping      Chinelo Liso   
...              ...        ...                         ...               ...   
100994         69996 2019-12-26  Center Shopping Uberlândia    Short Listrado   
100995         69996 2019-12-26  Center Shopping Uberlândia           Mochila   
100996         69996 2019-12-26  Center Shopping Uberlândia  Pulseira Estampa   
100997         69997 2019-12-26           Ribeirão Shopping   Camisa Listrado   
100998         69997 2019-12-26           Ribeirão Shopping       Short Linho   

        Quantidade  Valor Unitário  Valor Final  Comissao  
0                1             358          358     17.90  
1                2             180          360     18.00  
2                1             368          368     18.40  
3                3             200          600     30.00  
4                1              71           71      3.55  
...            ...             ...          ...       ...  
100994           2             102          204       NaN  
100995           4             270         1080       NaN  
100996           1              87           87       NaN  
100997           1             108          108       NaN  
100998           2             133          266       NaN  

[100999 rows x 8 columns]

### P16 - Prencha os valores vazios da coluna "Comissao" com a média do seus valores

In [37]:
vendasTot['Comissao'].fillna(vendasTot['Comissao'].mean(), inplace=True)
vendasTot

Código Venda       Data                     ID Loja           Produto  \
0                  1 2019-01-01          Iguatemi Esplanada    Sapato Estampa   
1                  1 2019-01-01          Iguatemi Esplanada          Camiseta   
2                  1 2019-01-01          Iguatemi Esplanada     Sapato Xadrez   
3                  2 2019-01-02              Norte Shopping           Relógio   
4                  2 2019-01-02              Norte Shopping      Chinelo Liso   
...              ...        ...                         ...               ...   
100994         69996 2019-12-26  Center Shopping Uberlândia    Short Listrado   
100995         69996 2019-12-26  Center Shopping Uberlândia           Mochila   
100996         69996 2019-12-26  Center Shopping Uberlândia  Pulseira Estampa   
100997         69997 2019-12-26           Ribeirão Shopping   Camisa Listrado   
100998         69997 2019-12-26           Ribeirão Shopping       Short Linho   

        Quantidade  Valor Unitário  Valor Final   Comissao  
0                1             358          358  17.900000  
1                2             180          360  18.000000  
2                1             368          368  18.400000  
3                3             200          600  30.000000  
4                1              71           71   3.550000  
...            ...             ...          ...        ...  
100994           2             102          204  20.743133  
100995           4             270         1080  20.743133  
100996           1              87           87  20.743133  
100997           1             108          108  20.743133  
100998           2             133          266  20.743133  

[100999 rows x 8 columns]

### P17 - Quantas transações por loja?

In [38]:
vendasTot['ID Loja'].value_counts()

Shopping Vila Velha                 4234
Palladium Shopping Curitiba         4210
Norte Shopping                      4179
Rio Mar Shopping Fortaleza          4118
Bourbon Shopping SP                 4116
Iguatemi Campinas                   4108
Rio Mar Recife                      4099
Shopping Center Leste Aricanduva    4093
Shopping SP Market                  4080
Shopping Ibirapuera                 4051
Novo Shopping Ribeirão Preto        4049
Ribeirão Shopping                   4048
Salvador Shopping                   4030
Shopping Center Interlagos          4021
Shopping Iguatemi Fortaleza         4021
Center Shopping Uberlândia          4013
Shopping Eldorado                   4002
Passei das Águas Shopping           4000
Shopping União de Osasco            3995
Iguatemi Esplanada                  3980
Shopping Barra                      3962
Shopping Morumbi                    3960
Parque Dom Pedro Shopping           3902
Shopping Recife                     3891
Shopping Midway 

### P18 - Faturamento total em todas as lojas por produto?

In [39]:
vendasTot[['Produto', 'Valor Final']].groupby(by='Produto').sum()

Valor Final
Produto                      
Bermuda                272250
Bermuda Estampa        291694
Bermuda Linho          394680
Bermuda Liso           275692
Bermuda Listrado       293237
...                       ...
Tênis Estampa          457728
Tênis Linho            538608
Tênis Liso             474544
Tênis Listrado         481032
Tênis Xadrez           459725

[120 rows x 1 columns]

## Tratamento dos dados (gerentes)

In [40]:
# carrega dados
gerentes = pd.read_excel('Gerentes.xlsx')
gerentes

ID Loja   Gerente
0                 Iguatemi Esplanada  Salvador
1                     Norte Shopping     Joana
2         Rio Mar Shopping Fortaleza     Maria
3                     Shopping Barra   Letícia
4                Shopping Ibirapuera    Helena
5                  Iguatemi Campinas     Luiza
6   Shopping Center Leste Aricanduva     Julia
7          Passei das Águas Shopping    Nicole
8                    Shopping Recife     André
9               Shopping Midway Mall      João
10               Bourbon Shopping SP    Marcus
11        Shopping Center Interlagos    Amanda
12         Parque Dom Pedro Shopping     Carol
13        Center Shopping Uberlândia  Andressa
14          Shopping União de Osasco     Lilia
15                 Shopping Eldorado     Paulo
16               Shopping Vila Velha     Pablo
17      Novo Shopping Ribeirão Preto     Diana
18                    Rio Mar Recife       Ana
19       Palladium Shopping Curitiba    Flavia
20                Shopping SP Market  Fernanda
21                 Ribeirão Shopping     Fábio
22       Shopping Iguatemi Fortaleza      Joel
23                  Shopping Morumbi    Gisele
24                 Salvador Shopping    Mateus

### P19 - Junte as tabelas gerentes e vendas (o 2019 inteiro)

In [41]:
venTotGer = vendasTot.merge(right=gerentes, how='inner', on='ID Loja')
venTotGer

Código Venda       Data             ID Loja                 Produto  \
0                  1 2019-01-01  Iguatemi Esplanada          Sapato Estampa   
1                  1 2019-01-01  Iguatemi Esplanada                Camiseta   
2                  1 2019-01-01  Iguatemi Esplanada           Sapato Xadrez   
3                 21 2019-01-02  Iguatemi Esplanada  Camisa Gola V Listrado   
4                 34 2019-01-02  Iguatemi Esplanada         Sapato Listrado   
...              ...        ...                 ...                     ...   
100994         69846 2019-12-25   Salvador Shopping           Short Estampa   
100995         69846 2019-12-25   Salvador Shopping           Tênis Estampa   
100996         69850 2019-12-25   Salvador Shopping           Calça Estampa   
100997         69972 2019-12-26   Salvador Shopping              Terno Liso   
100998         69972 2019-12-26   Salvador Shopping          Sapato Estampa   

        Quantidade  Valor Unitário  Valor Final   Comissao   Gerente  
0                1             358          358  17.900000  Salvador  
1                2             180          360  18.000000  Salvador  
2                1             368          368  18.400000  Salvador  
3                2             116          232  11.600000  Salvador  
4                1             363          363  18.150000  Salvador  
...            ...             ...          ...        ...       ...  
100994           2              96          192  20.743133    Mateus  
100995           5             256         1280  20.743133    Mateus  
100996           4             177          708  20.743133    Mateus  
100997           3             720         2160  20.743133    Mateus  
100998           5             358         1790  20.743133    Mateus  

[100999 rows x 9 columns]

### P20 - Quais são os top 5 gerentes que vederam mais por loja?

In [44]:
venTotGer[['ID Loja', 'Gerente', 'Valor Final']].groupby(by=['ID Loja', 'Gerente']).sum().sort_values(by='Valor Final', ascending=False)

Valor Final
ID Loja                          Gerente              
Iguatemi Campinas                Luiza         1762419
Shopping Vila Velha              Pablo         1731167
Bourbon Shopping SP              Marcus        1726110
Rio Mar Recife                   Ana           1722766
Shopping SP Market               Fernanda      1721763
Palladium Shopping Curitiba      Flavia        1721120
Norte Shopping                   Joana         1711968
Ribeirão Shopping                Fábio         1707122
Iguatemi Esplanada               Salvador      1699681
Rio Mar Shopping Fortaleza       Maria         1698430
Shopping Center Leste Aricanduva Julia         1682870
Novo Shopping Ribeirão Preto     Diana         1678225
Shopping Iguatemi Fortaleza      Joel          1674824
Center Shopping Uberlândia       Andressa      1668921
Shopping União de Osasco         Lilia         1663770
Shopping Barra                   Letícia       1653103
Passei das Águas Shopping        Nicole        1649014
Shopping Eldorado                Paulo         1648111
Salvador Shopping                Mateus        1646739
Shopping Center Interlagos       Amanda        1640486
Shopping Ibirapuera              Helena        1635060
Parque Dom Pedro Shopping        Carol         1631630
Shopping Recife                  André         1624879
Shopping Midway Mall             João          1590441
Shopping Morumbi                 Gisele        1586444